Automattic/wp-calypso
# Make `SiteFavicon` a shared component

Related to 
- https://github.com/Automattic/dotcom-forge/issues/10138
- https://github.com/Automattic/wp-calypso/pull/97635, https://github.com/Automattic/wp-calypso/pull/97593, https://github.com/Automattic/wp-calypso/pull/97542


## Proposed Changes

This PR makes `SiteFavicon` a shared component, extracting from A4A. 

## Why are these changes being made?


- https://github.com/Automattic/dotcom-forge/issues/10138
- Please refer to the Project Thread pbxlJb-6J5-p2 and the proposal post pfYzsZ-Y4-p2 for more context.

## Testing Instructions



both on Calypso and A4A Live

- Go to /site 
- Click any site 
- Ensure nothing breaks 

## Pre-merge Checklist
